src/analyzer/plugins/analyzerqueenmarybeats.cpp

```diff
@@ -70,13 +70,14 @@ bool AnalyzerQueenMaryBeats::finalize() {
     }
 
     std::vector<double> df;
-    std::vector<double> beatPeriod(nonZeroCount);
+    std::vector<double> beatPeriod;
     std::vector<double> tempi;
 
     df.reserve(nonZeroCount);
-
-    for (int i = 0; i < nonZeroCount; ++i) {
+    // no idea why skip first 2 results, but that's how vamp does and works best this way
```

Comment:
Do the `df` and `beatPeriod` vectors store correlated information and must be of the same size? I think that was the reason why you decided to populate them piece by piece in the same loop. But then we should better use a single vector and introduce a struct that subsumes the individual parameters.

I haven't checked how much work this refactoring would be, just a thought.

Reply:
Not a lot of work actually, but since these are declared just a few lines above where they are used and not used anywhere else, IMHO will have no impact on the code cleanliness and so it's not worth the overhead and boilerplate.
On the first look, the library code is a bit obscure and to be honest I don't fully comprehend how they are related.
So I didn't feel comfortable to make any alterations. I populated them that way because that's how it's done in the original vamp plugin and I wanted to be on the safe side.

Reply:
(+1 to not changing our local fork of qm-dsp much)

Reply:
This is our own analyzer code and not forked code.

Reply:
I don't see any copyright mentions or references to original code? That's why I assumed this is Mixxx code.

Reply:
Oh it is, but they are separate vectors because the QM-dsp library takes them as separate vectors. I thought you were proposing to change their interface to take a `vector<struct>`.